File: packages/philjs-tauri/src/window.rs
//! Window management for PhilJS Tauri

use crate::TauriError;
use tauri::{AppHandle, WebviewWindow, Wry};

/// Window wrapper
#[derive(Clone)]
pub struct Window {
    inner: WebviewWindow,
}

impl Window {
    pub fn new(window: WebviewWindow) -> Self {
        Window { inner: window }
    }

    /// Get the window label
    pub fn label(&self) -> &str {
        self.inner.label()
    }

    /// Show the window
    pub fn show(&self) -> Result<(), TauriError> {
        self.inner.show().map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Hide the window
    pub fn hide(&self) -> Result<(), TauriError> {
        self.inner.hide().map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Close the window
    pub fn close(&self) -> Result<(), TauriError> {
        self.inner.close().map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Minimize the window
    pub fn minimize(&self) -> Result<(), TauriError> {
        self.inner.minimize().map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Maximize the window
    pub fn maximize(&self) -> Result<(), TauriError> {
        self.inner.maximize().map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Unmaximize the window
    pub fn unmaximize(&self) -> Result<(), TauriError> {
        self.inner.unmaximize().map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Toggle maximize
    pub fn toggle_maximize(&self) -> Result<(), TauriError> {
        if self.is_maximized()? {
            self.unmaximize()
        } else {
            self.maximize()
        }
    }

    /// Check if maximized
    pub fn is_maximized(&self) -> Result<bool, TauriError> {
        self.inner.is_maximized().map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Check if minimized
    pub fn is_minimized(&self) -> Result<bool, TauriError> {
        self.inner.is_minimized().map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Check if visible
    pub fn is_visible(&self) -> Result<bool, TauriError> {
        self.inner.is_visible().map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Check if focused
    pub fn is_focused(&self) -> Result<bool, TauriError> {
        self.inner.is_focused().map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Set focus
    pub fn set_focus(&self) -> Result<(), TauriError> {
        self.inner.set_focus().map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Set title
    pub fn set_title(&self, title: &str) -> Result<(), TauriError> {
        self.inner.set_title(title).map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Set size
    pub fn set_size(&self, width: f64, height: f64) -> Result<(), TauriError> {
        use tauri::LogicalSize;
        self.inner
            .set_size(LogicalSize::new(width, height))
            .map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Set min size
    pub fn set_min_size(&self, width: Option<f64>, height: Option<f64>) -> Result<(), TauriError> {
        use tauri::LogicalSize;
        let size = match (width, height) {
            (Some(w), Some(h)) => Some(LogicalSize::new(w, h)),
            _ => None,
        };
        self.inner
            .set_min_size(size)
            .map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Set max size
    pub fn set_max_size(&self, width: Option<f64>, height: Option<f64>) -> Result<(), TauriError> {
        use tauri::LogicalSize;
        let size = match (width, height) {
            (Some(w), Some(h)) => Some(LogicalSize::new(w, h)),
            _ => None,
        };
        self.inner
            .set_max_size(size)
            .map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Set position
    pub fn set_position(&self, x: f64, y: f64) -> Result<(), TauriError> {
        use tauri::LogicalPosition;
        self.inner
            .set_position(LogicalPosition::new(x, y))
            .map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Center the window
    pub fn center(&self) -> Result<(), TauriError> {
        self.inner.center().map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Set fullscreen
    pub fn set_fullscreen(&self, fullscreen: bool) -> Result<(), TauriError> {
        self.inner
            .set_fullscreen(fullscreen)
            .map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Set always on top
    pub fn set_always_on_top(&self, always_on_top: bool) -> Result<(), TauriError> {
        self.inner
            .set_always_on_top(always_on_top)
            .map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Set resizable
    pub fn set_resizable(&self, resizable: bool) -> Result<(), TauriError> {
        self.inner
            .set_resizable(resizable)
            .map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Set decorations
    pub fn set_decorations(&self, decorations: bool) -> Result<(), TauriError> {
        self.inner
            .set_decorations(decorations)
            .map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Open dev tools
    #[cfg(debug_assertions)]
    pub fn open_devtools(&self) {
        self.inner.open_devtools();
    }

    /// Close dev tools
    #[cfg(debug_assertions)]
    pub fn close_devtools(&self) {
        self.inner.close_devtools();
    }

    /// Check if devtools open
    #[cfg(debug_assertions)]
    pub fn is_devtools_open(&self) -> bool {
        self.inner.is_devtools_open()
    }

    /// Emit event to window
    pub fn emit(&self, event: &str, payload: impl serde::Serialize) -> Result<(), TauriError> {
        self.inner
            .emit(event, payload)
            .map_err(|e| TauriError::Window(e.to_string()))
    }

    /// Listen to events from window
    pub fn listen<F>(&self, event: &str, handler: F) -> tauri::EventId
    where
        F: Fn(tauri::Event) + Send + 'static,
    {
        self.inner.listen(event, handler)
    }

    /// Unlisten to event
    pub fn unlisten(&self, id: tauri::EventId) {
        self.inner.unlisten(id);
    }
}

/// Window configuration
#[derive(Debug, Clone)]
pub struct WindowConfig {
    pub label: String,
    pub title: String,
    pub url: String,
    pub width: f64,
    pub height: f64,
    pub min_width: Option<f64>,
    pub min_height: Option<f64>,
    pub max_width: Option<f64>,
    pub max_height: Option<f64>,
    pub x: Option<f64>,
    pub y: Option<f64>,
    pub center: bool,
    pub resizable: bool,
    pub fullscreen: bool,
    pub decorations: bool,
    pub transparent: bool,
    pub always_on_top: bool,
    pub visible: bool,
    pub focused: bool,
}

impl Default for WindowConfig {
    fn default() -> Self {
        WindowConfig {
            label: "main".to_string(),
            title: "PhilJS App".to_string(),
            url: "/".to_string(),
            width: 1024.0,
            height: 768.0,
            min_width: Some(400.0),
            min_height: Some(300.0),
            max_width: None,
            max_height: None,
            x: None,
            y: None,
            center: true,
            resizable: true,
            fullscreen: false,
            decorations: true,
            transparent: false,
            always_on_top: false,
            visible: true,
            focused: true,
        }
    }
}

impl WindowConfig {
    pub fn new(label: impl Into<String>) -> Self {
        WindowConfig {
            label: label.into(),
            ..Default::default()
        }
    }

    pub fn title(mut self, title: impl Into<String>) -> Self {
        self.title = title.into();
        self
    }

    pub fn url(mut self, url: impl Into<String>) -> Self {
        self.url = url.into();
        self
    }

    pub fn size(mut self, width: f64, height: f64) -> Self {
        self.width = width;
        self.height = height;
        self
    }

    pub fn min_size(mut self, width: f64, height: f64) -> Self {
        self.min_width = Some(width);
        self.min_height = Some(height);
        self
    }

    pub fn max_size(mut self, width: f64, height: f64) -> Self {
        self.max_width = Some(width);
        self.max_height = Some(height);
        self
    }

    pub fn position(mut self, x: f64, y: f64) -> Self {
        self.x = Some(x);
        self.y = Some(y);
        self.center = false;
        self
    }

    pub fn center(mut self) -> Self {
        self.center = true;
        self.x = None;
        self.y = None;
        self
    }

    pub fn resizable(mut self, resizable: bool) -> Self {
        self.resizable = resizable;
        self
    }

    pub fn fullscreen(mut self) -> Self {
        self.fullscreen = true;
        self
    }

    pub fn frameless(mut self) -> Self {
        self.decorations = false;
        self
    }

    pub fn transparent(mut self) -> Self {
        self.transparent = true;
        self
    }

    pub fn always_on_top(mut self) -> Self {
        self.always_on_top = true;
        self
    }

    pub fn hidden(mut self) -> Self {
        self.visible = false;
        self
    }
}

/// Window builder
pub struct WindowBuilder<'a> {
    handle: &'a AppHandle<Wry>,
    config: WindowConfig,
}

impl<'a> WindowBuilder<'a> {
    pub fn new(handle: &'a AppHandle<Wry>, config: WindowConfig) -> Self {
        WindowBuilder { handle, config }
    }

    pub fn build(self) -> Result<Window, TauriError> {
        use tauri::{LogicalPosition, LogicalSize, WebviewUrl, WebviewWindowBuilder};

        let mut builder = WebviewWindowBuilder::new(
            self.handle,
            &self.config.label,
            WebviewUrl::App(self.config.url.into()),
        )
        .title(&self.config.title)
        .inner_size(self.config.width, self.config.height)
        .resizable(self.config.resizable)
        .fullscreen(self.config.fullscreen)
        .decorations(self.config.decorations)
        .transparent(self.config.transparent)
        .always_on_top(self.config.always_on_top)
        .visible(self.config.visible)
        .focused(self.config.focused);

        if let (Some(min_w), Some(min_h)) = (self.config.min_width, self.config.min_height) {
            builder = builder.min_inner_size(min_w, min_h);
        }

        if let (Some(max_w), Some(max_h)) = (self.config.max_width, self.config.max_height) {
            builder = builder.max_inner_size(max_w, max_h);
        }

        if self.config.center {
            builder = builder.center();
        } else if let (Some(x), Some(y)) = (self.config.x, self.config.y) {
            builder = builder.position(x, y);
        }

        let window = builder
            .build()
            .map_err(|e| TauriError::Window(e.to_string()))?;

        Ok(Window::new(window))
    }
}

File: packages/philjs-desktop/src/window.ts
/**
 * Window Management for PhilJS Desktop
 */

import { isTauri } from './tauri/context.js';
import { listen, emit, TauriEvents } from './tauri/events.js';
import type { UnlistenFn, Event } from './tauri/types.js';

// Window types
export interface WindowOptions {
  /** Window label (unique identifier) */
  label?: string;
  /** Window title */
  title?: string;
  /** Window width */
  width?: number;
  /** Window height */
  height?: number;
  /** Minimum width */
  minWidth?: number;
  /** Minimum height */
  minHeight?: number;
  /** Maximum width */
  maxWidth?: number;
  /** Maximum height */
  maxHeight?: number;
  /** X position */
  x?: number;
  /** Y position */
  y?: number;
  /** Center window on screen */
  center?: boolean;
  /** Resizable */
  resizable?: boolean;
  /** Decorations (title bar, etc.) */
  decorations?: boolean;
  /** Always on top */
  alwaysOnTop?: boolean;
  /** Fullscreen */
  fullscreen?: boolean;
  /** Visible on creation */
  visible?: boolean;
  /** Focus on creation */
  focus?: boolean;
  /** Transparent */
  transparent?: boolean;
  /** URL to load */
  url?: string;
  /** Parent window label */
  parent?: string;
  /** Skip taskbar */
  skipTaskbar?: boolean;
}

export interface WindowSize {
  width: number;
  height: number;
}

export interface WindowPosition {
  x: number;
  y: number;
}

export interface PhysicalSize {
  width: number;
  height: number;
}

export interface PhysicalPosition {
  x: number;
  y: number;
}

export interface Monitor {
  name: string | null;
  size: PhysicalSize;
  position: PhysicalPosition;
  scaleFactor: number;
}

export interface WindowState {
  label: string;
  title: string;
  isVisible: boolean;
  isFullscreen: boolean;
  isMaximized: boolean;
  isMinimized: boolean;
  isFocused: boolean;
  isDecorated: boolean;
  isResizable: boolean;
  isAlwaysOnTop: boolean;
  size: WindowSize;
  position: WindowPosition;
}

// Window manager state
const windowInstances = new Map<string, WindowHandle>();
let currentWindowLabel: string | null = null;

/**
 * Window handle class
 */
export class WindowHandle {
  private label: string;
  private tauriWindow: any = null;

  constructor(label: string, tauriWindow?: any) {
    this.label = label;
    this.tauriWindow = tauriWindow;
  }

  getLabel(): string {
    return this.label;
  }

  async close(): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.close();
    windowInstances.delete(this.label);
  }

  async destroy(): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.destroy();
    windowInstances.delete(this.label);
  }

  async minimize(): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.minimize();
  }

  async maximize(): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.maximize();
  }

  async unmaximize(): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.unmaximize();
  }

  async toggleMaximize(): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.toggleMaximize();
  }

  async show(): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.show();
  }

  async hide(): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.hide();
  }

  async setTitle(title: string): Promise<void> {
    if (!isTauri()) {
      document.title = title;
      return;
    }
    await this.tauriWindow?.setTitle(title);
  }

  async setSize(width: number, height: number): Promise<void> {
    if (!isTauri()) return;
    const { LogicalSize } = await import('@tauri-apps/api/dpi');
    await this.tauriWindow?.setSize(new LogicalSize(width, height));
  }

  async setMinSize(width: number, height: number): Promise<void> {
    if (!isTauri()) return;
    const { LogicalSize } = await import('@tauri-apps/api/dpi');
    await this.tauriWindow?.setMinSize(new LogicalSize(width, height));
  }

  async setMaxSize(width: number, height: number): Promise<void> {
    if (!isTauri()) return;
    const { LogicalSize } = await import('@tauri-apps/api/dpi');
    await this.tauriWindow?.setMaxSize(new LogicalSize(width, height));
  }

  async setPosition(x: number, y: number): Promise<void> {
    if (!isTauri()) return;
    const { LogicalPosition } = await import('@tauri-apps/api/dpi');
    await this.tauriWindow?.setPosition(new LogicalPosition(x, y));
  }

  async center(): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.center();
  }

  async setFullscreen(enabled: boolean): Promise<void> {
    if (!isTauri()) {
      if (enabled) {
        document.documentElement.requestFullscreen?.();
      } else {
        document.exitFullscreen?.();
      }
      return;
    }
    await this.tauriWindow?.setFullscreen(enabled);
  }

  async setAlwaysOnTop(enabled: boolean): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.setAlwaysOnTop(enabled);
  }

  async setResizable(enabled: boolean): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.setResizable(enabled);
  }

  async setDecorations(enabled: boolean): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.setDecorations(enabled);
  }

  async setFocus(): Promise<void> {
    if (!isTauri()) {
      window.focus();
      return;
    }
    await this.tauriWindow?.setFocus();
  }

  async setSkipTaskbar(skip: boolean): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.setSkipTaskbar(skip);
  }

  async requestUserAttention(type: 'critical' | 'informational' | null): Promise<void> {
    if (!isTauri()) return;
    const { UserAttentionType } = await import('@tauri-apps/api/window');
    const attentionType = type === 'critical'
      ? UserAttentionType.Critical
      : type === 'informational'
        ? UserAttentionType.Informational
        : null;
    await this.tauriWindow?.requestUserAttention(attentionType);
  }

  async getSize(): Promise<WindowSize> {
    if (!isTauri()) {
      return { width: window.innerWidth, height: window.innerHeight };
    }
    const size = await this.tauriWindow?.innerSize();
    return { width: size?.width || 0, height: size?.height || 0 };
  }

  async getPosition(): Promise<WindowPosition> {
    if (!isTauri()) {
      return { x: window.screenX, y: window.screenY };
    }
    const pos = await this.tauriWindow?.outerPosition();
    return { x: pos?.x || 0, y: pos?.y || 0 };
  }

  async isVisible(): Promise<boolean> {
    if (!isTauri()) return document.visibilityState === 'visible';
    return this.tauriWindow?.isVisible() ?? true;
  }

  async isMaximized(): Promise<boolean> {
    if (!isTauri()) return false;
    return this.tauriWindow?.isMaximized() ?? false;
  }

  async isMinimized(): Promise<boolean> {
    if (!isTauri()) return false;
    return this.tauriWindow?.isMinimized() ?? false;
  }

  async isFullscreen(): Promise<boolean> {
    if (!isTauri()) return !!document.fullscreenElement;
    return this.tauriWindow?.isFullscreen() ?? false;
  }

  async isFocused(): Promise<boolean> {
    if (!isTauri()) return document.hasFocus();
    return this.tauriWindow?.isFocused() ?? false;
  }

  async getState(): Promise<WindowState> {
    const [size, position, isVisible, isMaximized, isMinimized, isFullscreen, isFocused] =
      await Promise.all([
        this.getSize(),
        this.getPosition(),
        this.isVisible(),
        this.isMaximized(),
        this.isMinimized(),
        this.isFullscreen(),
        this.isFocused(),
      ]);

    return {
      label: this.label,
      title: document.title,
      isVisible,
      isMaximized,
      isMinimized,
      isFullscreen,
      isFocused,
      isDecorated: true,
      isResizable: true,
      isAlwaysOnTop: false,
      size,
      position,
    };
  }

  async onResize(callback: (size: WindowSize) => void): Promise<UnlistenFn> {
    return listen(TauriEvents.WINDOW_RESIZED, (e: Event<WindowSize>) => {
      callback(e.payload as WindowSize);
    });
  }

  async onMove(callback: (position: WindowPosition) => void): Promise<UnlistenFn> {
    return listen(TauriEvents.WINDOW_MOVED, (e: Event<WindowPosition>) => {
      callback(e.payload as WindowPosition);
    });
  }

  async onFocus(callback: () => void): Promise<UnlistenFn> {
    return listen(TauriEvents.WINDOW_FOCUS, callback);
  }

  async onBlur(callback: () => void): Promise<UnlistenFn> {
    return listen(TauriEvents.WINDOW_BLUR, callback);
  }

  async onCloseRequested(callback: () => void): Promise<UnlistenFn> {
    return listen(TauriEvents.WINDOW_CLOSE_REQUESTED, callback);
  }

  async startDragging(): Promise<void> {
    if (!isTauri()) return;
    await this.tauriWindow?.startDragging();
  }
}

/**
 * Create a new window
 */
export async function createWindow(options: WindowOptions = {}): Promise<WindowHandle> {
  const label = options.label || `window-${Date.now()}`;

  if (windowInstances.has(label)) {
    throw new Error(`Window with label "${label}" already exists`);
  }

  if (!isTauri()) {
    // In browser, create a new popup window
    const features = [
      `width=${options.width || 800}`,
      `height=${options.height || 600}`,
      options.x !== undefined ? `left=${options.x}` : '',
      options.y !== undefined ? `top=${options.y}` : '',
      `resizable=${options.resizable !== false ? 'yes' : 'no'}`,
    ].filter(Boolean).join(',');

    const popup = window.open(options.url || '', label, features);
    if (popup && options.title) {
      popup.document.title = options.title;
    }

    const handle = new WindowHandle(label);
    windowInstances.set(label, handle);
    return handle;
  }

  const { WebviewWindow } = await import('@tauri-apps/api/webviewWindow');

  const webview = new WebviewWindow(label, {
    title: options.title,
    width: options.width,
    height: options.height,
    minWidth: options.minWidth,
    minHeight: options.minHeight,
    maxWidth: options.maxWidth,
    maxHeight: options.maxHeight,
    x: options.x,
    y: options.y,
    center: options.center,
    resizable: options.resizable,
    decorations: options.decorations,
    alwaysOnTop: options.alwaysOnTop,
    fullscreen: options.fullscreen,
    visible: options.visible,
    focus: options.focus,
    transparent: options.transparent,
    url: options.url,
    parent: options.parent,
    skipTaskbar: options.skipTaskbar,
  });

  // Wait for window to be created
  await new Promise<void>((resolve, reject) => {
    webview.once('tauri://created', () => resolve());
    webview.once('tauri://error', (e: any) => reject(e));
  });

  const handle = new WindowHandle(label, webview);
  windowInstances.set(label, handle);

  return handle;
}

/**
 * Get the current window
 */
export async function getCurrentWindow(): Promise<WindowHandle> {
  if (!isTauri()) {
    const label = 'main';
    if (!windowInstances.has(label)) {
      windowInstances.set(label, new WindowHandle(label));
    }
    return windowInstances.get(label)!;
  }

  const { getCurrentWindow: getTauriWindow } = await import('@tauri-apps/api/window');
  const tauriWindow = getTauriWindow();
  const label = tauriWindow.label;

  if (!windowInstances.has(label)) {
    windowInstances.set(label, new WindowHandle(label, tauriWindow));
  }

  currentWindowLabel = label;
  return windowInstances.get(label)!;
}

/**
 * Hook to get the current window
 */
export function useWindow(): WindowHandle {
  // This would integrate with reactive system
  // For now, return a placeholder that lazy-loads
  const label = currentWindowLabel || 'main';
  if (!windowInstances.has(label)) {
    windowInstances.set(label, new WindowHandle(label));
  }
  return windowInstances.get(label)!;
}

/**
 * Get all windows
 */
export async function getAllWindows(): Promise<WindowHandle[]> {
  if (!isTauri()) {
    return Array.from(windowInstances.values());
  }

  const { getAllWindows: getTauriWindows } = await import('@tauri-apps/api/window');
  const tauriWindows = await getTauriWindows();

  for (const tw of tauriWindows) {
    if (!windowInstances.has(tw.label)) {
      windowInstances.set(tw.label, new WindowHandle(tw.label, tw));
    }
  }

  return Array.from(windowInstances.values());
}

/**
 * Get a window by label
 */
export async function getWindow(label: string): Promise<WindowHandle | null> {
  if (windowInstances.has(label)) {
    return windowInstances.get(label)!;
  }

  if (!isTauri()) {
    return null;
  }

  const { WebviewWindow } = await import('@tauri-apps/api/webviewWindow');
  const tw = await WebviewWindow.getByLabel(label);

  if (tw) {
    const handle = new WindowHandle(label, tw);
    windowInstances.set(label, handle);
    return handle;
  }

  return null;
}

// Convenience functions for current window
export async function closeWindow(): Promise<void> {
  const win = await getCurrentWindow();
  await win.close();
}

export async function minimizeWindow(): Promise<void> {
  const win = await getCurrentWindow();
  await win.minimize();
}

export async function maximizeWindow(): Promise<void> {
  const win = await getCurrentWindow();
  await win.maximize();
}

export async function setTitle(title: string): Promise<void> {
  const win = await getCurrentWindow();
  await win.setTitle(title);
}

export async function setSize(width: number, height: number): Promise<void> {
  const win = await getCurrentWindow();
  await win.setSize(width, height);
}

export async function setFullscreen(enabled: boolean): Promise<void> {
  const win = await getCurrentWindow();
  await win.setFullscreen(enabled);
}

export async function setAlwaysOnTop(enabled: boolean): Promise<void> {
  const win = await getCurrentWindow();
  await win.setAlwaysOnTop(enabled);
}

export async function center(): Promise<void> {
  const win = await getCurrentWindow();
  await win.center();
}

export async function setPosition(x: number, y: number): Promise<void> {
  const win = await getCurrentWindow();
  await win.setPosition(x, y);
}

/**
 * Get primary monitor info
 */
export async function getPrimaryMonitor(): Promise<Monitor | null> {
  if (!isTauri()) return null;

  const { primaryMonitor } = await import('@tauri-apps/api/window');
  const monitor = await primaryMonitor();

  if (!monitor) return null;

  return {
    name: monitor.name,
    size: monitor.size,
    position: monitor.position,
    scaleFactor: monitor.scaleFactor,
  };
}

/**
 * Get all monitors
 */
export async function getAllMonitors(): Promise<Monitor[]> {
  if (!isTauri()) return [];

  const { availableMonitors } = await import('@tauri-apps/api/window');
  const monitors = await availableMonitors();

  return monitors.map(m => ({
    name: m.name,
    size: m.size,
    position: m.position,
    scaleFactor: m.scaleFactor,
  }));
}
